packages/bot/src/stores/quick-strategy-store.js

```diff
@@ -98,41 +110,128 @@ export default class QuickStrategyStore {
+                }
+            }
+    
+            if (value === '') {
+                errors[key] = translate('Field cannot be empty');
+            }
+        });
+
+        const { min, max } = this.duration_dropdown.filter(duration => duration.unit === values.duration_type)[0];
+        if (values.duration < min) {
+            errors.duration = `${translate('Minimum duration :')} ${min}`;
+        }
+
+        if (values.duration > max) {
+            errors.duration = `${translate('Maximum duration :')} ${max}`;
+        }
+    
+        return errors;
+    };
+
     @action.bound
     async onMount() {
         const { active_symbols } = ApiHelper.instance;
         const market_options     = await active_symbols.getAssetOptions();
 
-        this.updateTradetypeDropdown();
+        await this.updateTradetypeDropdown();
+        await this.updateDurationDropdown();
+        await this.updateDurationValue();
             
         runInAction(() => {
             this.market_dropdown = market_options;
         });
     }
 
     @action.bound
-    async updateTradetypeDropdown(symbol = this.initial_values.symbol) {
+    // eslint-disable-next-line
+    async onChangeMarketDropdown(setFieldValue, value) {
+        setFieldValue('symbol', value);
+        await this.updateTradetypeDropdown(setFieldValue, value);
+
+        const first_trade_type_option = this.trade_type_dropdown[Object.keys(this.trade_type_dropdown)[0]][0].value;
+
+        setFieldValue('trade_type', first_trade_type_option);
+    }
+
+    @action.bound
+    async updateTradetypeDropdown(setFieldValue, symbol = this.initial_values.symbol) {
         const { contracts_for }  = ApiHelper.instance;
         const trade_type_options = await contracts_for.getGroupedTradeTypes(symbol);
+        const first_trade_type_option = trade_type_options[Object.keys(trade_type_options)[0]][0].value;
 
-        runInAction(() => {
-            this.trade_type_dropdown = trade_type_options;
-        });
+        runInAction(() => this.trade_type_dropdown = trade_type_options);
+
+        if (setFieldValue) {
+            setFieldValue('trade_type', first_trade_type_option);
+    
+            await this.updateDurationDropdown(setFieldValue, symbol, first_trade_type_option);
+        }
     }
 
     @action.bound
     // eslint-disable-next-line
-    async onChangeMarketDropdown(setFieldValue, value) {
-        setFieldValue('symbol', value);
-        await this.updateTradetypeDropdown(value);
+    async onChangeTradeTypeDropdown(setFieldValue, symbol, trade_type) {
+        setFieldValue('trade_type', trade_type);
 
-        const first_option = this.trade_type_dropdown[Object.keys(this.trade_type_dropdown)[0]][0].value;
-        setFieldValue('tradetype', first_option);
+        await this.updateDurationDropdown(setFieldValue, symbol, trade_type);
+    }
+
+    @action.bound
+    async updateDurationDropdown(
+        setFieldValue,
```

Comment:
Same here, since `setFieldValue` does not always have a value it would be good to mark this as an optional argument by supplying a default value of e.g. `null`.